src/main/java/io/github/mainstringargs/abstracts/websocket/client/AbstractWebsocketClientEndpoint.java

```diff
@@ -100,34 +98,28 @@ protected void onOpen(Session userSession) {
      * @param reason      the reason
      */
     protected void onClose(Session userSession, CloseReason reason) {
-        this.userSession = null;
-
-        LOGGER.debug("onClose " + userSession);
+        LOGGER.debug("onClose {}", userSession);
 
         if (!reason.getCloseCode().equals(CloseReason.CloseCodes.NORMAL_CLOSURE)) {
-            if (retryAttempts > 5) {
-                LOGGER.error("More than 5 attempts to reconnect were made.");
-                return;
-            }
-
-            LOGGER.info("Attempting a reconnect in 3 seconds.");
-            retryAttempts++;
```

Comment:
I removed retry logic because an application using this library can handle that if desired.

Reply:
Is there documentation someplace where a user can watch for these websocket events?  From my quick scan, these seems to be no way to know from the "client" side (application using the library) - it seems to be fire and forget.  It would be awesome if I am wrong though. :)

Reply:
Yes I can add an event that listeners can hook into. I'm also considering just adding the functionality back, but in a cleaner way. I will also add an `@OnError` for the websocket classes to report websocket related errors to listeners if desired.

Reply:
Exposing websocket events to the listeners sounds like the cleanest way to go about it.  This way, the user of the library can set it exactly what they want to do on these events.